icd/CMakeLists.txt

```diff
@@ -227,6 +227,13 @@ endif()
 ### PAL ########################################################################
 add_subdirectory(${XGL_PAL_PATH} ${PROJECT_BINARY_DIR}/pal)
 
+### XGL cache creator tool ####################################################
+if(ICD_BUILD_LLPC)
```

Comment:
This will build the cache creator tool every single time XGL builds against LLPC. Is this what XGL wants?

An alternative would be to create a separate option like:
```
option(XGL_BULD_CACHE_CREATOR_TOOL "Build cache creator tool" OFF)
```

Reply:
I want the cache creator tool to be compiled by default. This will make it less likely to get broken during downstream merges.

Reply:
The tool binary is quite small, it's ~11MB in debug mode on my machine.

Reply:
Adding @JonCampbell407

Reply:
The way you wrote your cmake isn't guaranteed anyway that this target will get built. You only wrote your cmake to guarantee your code is configured when ICD_BUILD_LLPC is true. Since there isn't a clear dependency on this new executable target.

If you've ever generated a visual studio project you should be familiar with the ALL_BUILD project:
![image](https://user-images.githubusercontent.com/74789911/101828493-a0931f80-3aff-11eb-9e71-ca725b823edf.png)

If you build your project this way it builds all available targets:
EX:
```
cmake --build build/
```
^ This way of building the project build all targets

If you build your project this way it builds only the targets (IE code) necessary to build xgl:
EX:
```
cmake --build build/ --target xgl
```

I'm not really sure which one XGL is using for CI purposes.

And since there isn't a clearly invoked dependency your target isn't guaranteed to be compiled each time.

If you want to add this as a clear build dependency you would use 'add_dependency'.
https://cmake.org/cmake/help/latest/command/add_dependencies.html?highlight=add_dependency

However as I understand it @JonCampbell407 doesn't want building the ICD to require building the cache creator tool.

Reply:
Cmake build documentation:
https://cmake.org/cmake/help/latest/manual/cmake.1.html#build-a-project

Reply:
Or of course you could invoke the actual build tool directly. I only presented the cmake ways of invoking the build step.

Reply:
I don't have a Windows machine and don't know how cmake interacts with Visual Studio.

When use the cmake invocation used by the CI builds (https://github.com/GPUOpen-Drivers/llpc/blob/339bb6335e02a5cc935e06604e5b42ab05622d89/docker/amdvlk.Dockerfile#L105), cache-creator gets built like I'd expect.

So if we decided to make it build every time, we would either have to explicitly request this target to be sure (`cmake --build . --target cache-creator`), or add a dependency `add_dependencies(cache-creator xgl)`?

Reply:
"So if we decided to make it build every time, we would either have to explicitly request this target to be sure (cmake --build . --target cache-creator), or add a dependency add_dependencies(cache-creator xgl)?"

Correct. At this point this is a decision for people on the AMD Vulkan team to decide on.
They pinged me about reviewing this code. But I was happy to help out :)

Reply:
I updated the cmake files to only compile cache creator when `XGL_BUILD_CACHE_CREATOR` is set. It will also check if LLPC is enabled and error out if not. I also made the `xgl` target to depend on `cache-creator`, so that (when enabled) it gets built when xgl is.